platforms/riscv32-cheri/features/BUILD.bazel

```diff
@@ -0,0 +1,65 @@
+load(
+    "//config:features.bzl",
+    "CPP_ALL_COMPILE_ACTIONS",
+    "C_ALL_COMPILE_ACTIONS",
+    "LD_ALL_ACTIONS",
+    "feature",
+    "feature_set",
+    "flag_group",
+    "flag_set",
+)
+
+package(default_visibility = ["//visibility:public"])
+
+feature(
+    name = "architecture",
+    enabled = True,
+    flag_sets = [
+        flag_set(
+            actions = CPP_ALL_COMPILE_ACTIONS + C_ALL_COMPILE_ACTIONS + LD_ALL_ACTIONS,
+            flag_groups = [
+                flag_group(
+                    flags = [
+                        "-march=ARCHITECTURE",
+                        "-mabi=ABI",
+                        "-mcmodel=CMODEL",
+                        "-mENDIAN-endian",
+                        "--target=riscv32-unknown-elf",
+                        "-Wl,-Lexternal/cheri_llvm_files/sdk/baremetal/baremetal-riscv32-purecap/riscv32-unknown-elf/lib/",
```

Comment:
This has been done in the new commit, but the substitution is in the device rather than the toolchain as a riscv64 device will probably want to look in a different directory for its libraries. I've also kept listify_flags despite having only one directory in case more are needed later. I've also changed the target to be a substitution (again done by the device).

Reply:
LGTM